client/src/components/content/Content.js

```diff
@@ -0,0 +1,73 @@
+import React, { Component } from 'react'
+import TopPanel from '../topPanel/TopPanel'
+import MessageList from '../message/MessageList'
+import { Loader } from '../ui'
+import { getReportsByTag } from '../../utils/serverApi'
+import './Content.scss'
+
+class Content extends Component {
+  state = {
+    emoji: {},
+    reports: [],
+    loading: true,
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.createState(nextProps)
+  }
+
+  componentWillMount() {
+    this.createState()
+  }
+
+  createState = async (props = this.props) => {
+    this.setState({
+      loading: true
+    })
+
+    const { tag } = props.match.params
+    let selectedTag = tag || props.defaultTag
+    selectedTag = selectedTag ? decodeURI(selectedTag) : selectedTag
+
+    const {reports, users, emoji, channels} = await getReportsByTag(selectedTag)
+    
+    this.setState({
+      reports,
+      users,
+      emoji,
+      channels,
+      selectedTag,
+      loading: props.loading,
+    })
+  }
+
+  render() {
+    const {emoji, reports, users, channels, selectedTag, loading} = this.state
+    console.log('XXX')
+    console.log(selectedTag)
+    console.log(selectedTag === undefined)
```

Comment:
Debugging stuff was left here.

Reply:
Thanks for pointing that out. Could you please either rewrite or fix this PR?

Reply:
:+1: Done.